mesonbuild/coredata.py

```diff
@@ -194,7 +195,7 @@ def __init__(self, options):
         self.compiler_options = {}
         self.base_options = {}
         self.cross_file = self.__load_cross_file(options.cross_file)
-        self.wrap_mode = options.wrap_mode
+        self.wrap_mode = options.wrap_mode if options.wrap_mode is not None else WrapMode.default
```

Comment:
Why is this needed? How is `None` different from `default`?

Reply:
Because we need to know if the user explicitly passed --wrap-mode=default or if he omitted the option from the command line. This is important in the next commit "mesonmain: Add --wipe argument".

Let's say user configure with `--wrap-mode=forcefallback`, the cmd_line.txt file will contain `wrap_mode=forcefallback`. Now the user wipe his build dir with `meson setup --wipe` we'll have options.wrap_mode=default, does it mean user ran the command `meson setup --wipe --wrap-mode=default` to explicitly override that value from cmd_line.txt, or does it mean he didn't pass any value and thus we should still used `wrap_mode=forcefallback`?

To fix that ambiguity, when user does not set --wrap-mode in the command line, options.wrap_mode is None.